pkg/validation/validation.go

```diff
@@ -377,24 +388,24 @@ func versionValidation(ucVersion string, cV *configv1.ClusterVersion, logger log
 	parsedUcVersion, err := semver.Parse(ucVersion)
 	if err != nil {
 		logger.Error(err, fmt.Sprintf("failed to parse upgrade config desired version %s as semver", ucVersion))
-		return false, false, err
+		return false, false, fmt.Errorf("failed to parse upgrade config desired version %s as semver: %v", ucVersion, err)
 	}
 
 	cvVersion, err := cv.GetCurrentVersion(cV)
 	if err != nil {
 		logger.Error(err, "failed to get current cluster version during validation")
-		return false, false, err
+		return false, false, fmt.Errorf("failed to get current cluster version during validation: %v", err)
 	}
 	parsedCvVersion, err := semver.Parse(cvVersion)
 	if err != nil {
-		logger.Error(err, fmt.Sprintf("Failed to parse current cluster version %s as semver", cvVersion))
-		return false, false, err
+		logger.Error(err, fmt.Sprintf("failed to parse current cluster version %s as semver", cvVersion))
+		return false, false, fmt.Errorf("failed to parse current cluster version %s as semver: %v", cvVersion, err)
 	}
 
 	// Compare versions to ascertain if upgrade should proceed.
 	versionComparison, err := compareVersions(parsedUcVersion, parsedCvVersion, logger)
 	if err != nil {
-		return false, false, err
+		return false, false, fmt.Errorf("failed to compare versions: %v", err)
```

Comment:
nit: if you use [`%w`][1] to format `err` in `Errorf`, you get nice [`Unwrap` support][2].

[1]: https://pkg.go.dev/fmt#Errorf
[2]: https://go.dev/blog/go1.13-errors

Reply:
Thanks! Fixed.